File: tests/fixtures/v2-upload-flow/positive/upload_flow.py
def direct(request, content):
    with open(request.FILES["upload"].name, "w") as output:
        output.write(content)


def propagated(request, content):
    requested = request.FILES["upload"].name
    alias = requested
    with open(alias, "w") as output:
        output.write(content)


def canonicalized(request, content):
    destination = os.path.realpath(request.FILES["upload"].name)
    with open(destination, "w") as output:
        output.write(content)


def valid_basename(filename):
    return os.path.basename(filename) == filename

File: tests/fixtures/v2-upload-flow/positive/upload-flow.ts
function direct(req: any, fs: any, content: any) {
  return fs.writeFileSync(req.file.originalname, content)
}

function propagated(req: any, fs: any, content: any) {
  const requested = req.file.originalname
  const alias = requested
  return fs.writeFileSync(alias, content)
}

function canonicalized(req: any, fs: any, path: any, content: any) {
  const destination = path.resolve(req.file.originalname)
  return fs.writeFileSync(destination, content)
}

function validBasename(filename: string, path: any) {
  return path.basename(filename) === filename
}
